### PR TITLE
Force contributors to provide a name for tickets

### PR DESCRIPTION
**Pull request by criticalAY**
# Description
Marking the name field as required when making a contribution as guest

Fixes: https://github.com/opencollective/opencollective/issues/6927

# Screenshots
image

### CODE REVIEW

**Comment by criticalAY** [comment]
I would like to know if I have to update the strings by myself or if the maintainers do it. And if I have to do it myself that what is the process.

**Review by Betree — CHANGES_REQUESTED** [review]
We want to enforce that only for event tickets, not all contributions

**Comment by Betree** [comment]
@criticalAY Any progress on that one?

**Comment by Betree** [comment]
> I would like to know if I have to update the strings by myself or if the maintainers do it. And if I have to do it myself that what is the process.

After updating the string, you need to run `npm run build:langs` to update the language files

**Comment by criticalAY** [comment]
I will fix it (busy at the moment)

**Comment by Betree** [comment]
> I will fix it (busy at the moment)

No problem! Just let us know if you don't feel like completing it, it's ok to drop an issue if you don't have the bandwidth to work on it anymore :slightly_smiling_face:

**Comment by criticalAY** [comment]
Hey I have added the functionality sorry for the delay I caused

File: components/contribution-flow/StepProfileGuestForm.js [hunk 1]
```diff
@@ -97,9 +98,13 @@ const StepProfileGuestForm = ({ stepDetails, onChange, data, isEmbed, onSignInCl
...
         labelFontWeight="700"
-        required={false}
+        required={tier.type === TierTypes.TICKET}
         hint={
-          <FormattedMessage defaultMessage="This is your display name or alias. Leave it in blank to appear as guest." />
+          tier.type === TierTypes.TICKET ? (
+            <FormattedMessage defaultMessage="This is your display name or alias." />
+          ) : (
+            <FormattedMessage defaultMessage="This is your display name or alias. Leave it in blank to appear as guest." />
+          )
```
[the remark this answers is not in the document]
Comment (criticalAY) [reply components/contribution-flow/StepProfileGuestForm.js:107]: There is an error when running the command in terminal

[the remark this answers is not in the document]
Comment (criticalAY) [reply components/contribution-flow/StepProfileGuestForm.js:107]: Something related to other file that I didn't even change

**Comment by criticalAY** [comment]
Hey! If it's possible to extract the strings using a workflow then please I would like that otherwise I would set up a linux on my system specially for this

**Comment by Betree** [comment]
@criticalAY Our codebase doesn't officially support Windows, though we'd be happy to merge any solution you have to this problem. Otherwise, we can run the command ourselves on the PR; just let me know if you want to do it that way.

**Review by Betree — COMMENTED** [review]
There's an issue in the proposed solution, mostly because the specs were not very explicit: `name` is your public name, the one displayed on the platform for your contribution. We don't want to enforce it because it's ok for people to contribute without displaying their real name to everyone.

So the field we really want to enforce is `legalName`, and therefore the fix should be in the `contributionRequiresLegalName` helper.

**Comment by criticalAY** [comment]
> There's an issue in the proposed solution, mostly because the specs were not very explicit: `name` is your public name, the one displayed on the platform for your contribution. We don't want to enforce it because it's ok for people to contribute without displaying their real name to everyone.
> 
> So the field we really want to enforce is `legalName`, and therefore the fix should be in the `contributionRequiresLegalName` helper.

Done

**Review by Betree — COMMENTED** [review]
Looking good! We just need to cleanup `components/contribution-flow/StepProfileGuestForm.js`, there's a leftover there (see CI results)

**Review by Betree — APPROVED** [review]

**Comment by Betree** [comment]
Linter is still failing, see https://github.com/opencollective/opencollective-frontend/actions/runs/6789314258/job/18474100699?pr=9193#step:4:23

**Comment by criticalAY** [comment]
> Linter is still failing, see https://github.com/opencollective/opencollective-frontend/actions/runs/6789314258/job/18474100699?pr=9193#step:4:23

I ran the lint fix this time it will not fail now

**Comment by Betree** [comment]
Thanks, merging now!

**Pull request merged**